docs/amr.md

```diff
@@ -30,7 +30,10 @@ In each refinement block, you are required to provide a ``method`` which is a st
 ### Predefined Criteria
 | Method | Description |
 |--------|-------------|
-| derivative_order_1 | ![formula](https://render.githubusercontent.com/render/math?math=\|dlnq\/dlnx\|), where q is the user selected variable |
+| derivative_order_1 | ![formula](https://render.githubusercontent.com/render/math?math=\|dlnq\/dlnx\|) |
+| derivative_order_2 | $$\frac{(\delta x_i)^2}{\|q\|} \left\| \frac{\partial^2 q}{\partial x_i^2} \right\|$$ Note that, as programmed, this quantity is bounded by [0,2). |
```

Comment:
I think there are some `| |` missing around the second deriv term.
Also, it'd be great if you could add a pointer to some doc (or document directly), why the the implementation differs from the definition here.
There are probably good reasons for it (and done in other codes) so this may help user to pick the right values based on the reference (e.g., the implementation in Pluto of the "second deriv" criterion is quite different from the one used here).

Reply:
@pgrete The absolute value signs around the second derivative are present in the doc.  If you look at the file diff. page you can select a button to show the rich text preview of the doc.

The coordinate index subscript is not intended to implicitly mean to sum on the repeated index, rather each coordinate direction is tested independently.  Admittedly, this code assumes Cartesian coordinates, though that assumption is also present in the first derivative refinement criterion.

Honestly, I don't know what's done in Pluto - I've never looked at the code.  The denominator is a measure of the conserved variable, it's

$$  \frac{|q_i| + 0.5 |q_{i-1} + q_{i+1}| }{2} $$

The reason I use this for normalizing the refinement criterion can be seen by considering limiting cases where say $q_i \le 0$ and $(q_{i-1} + q_{i+1}) > 0$.  This often happens for say the momentum or magnetic field components.  Normalizing the second derivative in this way results in a mathematically bounded expression between $[0, 2]$.  In the implementation, I followed the convention used in the first derivative refinement criterion to include a tiny floating point value which formally changes the bounding to $[0,2)$.  This is noted in the documentation.

Are you asking for something more formal as to why this expression is used, like a description of these details in the doc file?

Reply:
@pgrete I looked over the code again and it looks like to be consistent with the documentation I need an additional factor of 1/2.  Would you like me to add the factor of 1/2 to the documentation or write out explicitly the case for say the x coordinate so that it's clear.  In the end, I'm recommending second derivative error measure that's proportional to this

$$  \frac{ \| q_{i-1} + q_{i+1} - 2 q_i \| }{ \| q_{i-1} + q_{i+1} \| + 2 \| q_i \|} $$

In this form it's bounded by $[0,1]$.

Reply:
I revised the code and documentation to match this error criterion.

Reply:
> The absolute value signs around the second derivative are present in the doc. If you look at the file diff. page you can select a button to show the rich text preview of the doc.

I used the rich text version and it doesn't seem to properly render on my end:
![image](https://user-images.githubusercontent.com/22685568/200877106-f8853de1-2ecc-4448-bee5-e71a9c738bb0.png)
So that's probably the way it is.

Regarding the doc, it'd be great if you could add that you use a "weighted q" in the denominator (for reasons) just to make it transparent on how to get from the analytic to the discrete formula.
However, as this is just nitpicking I'm already approving so don't consider this blocking my approval.

Reply:
How odd - here's what I see
<img width="1083" alt="doc" src="https://user-images.githubusercontent.com/45148532/200898653-fa283671-71b8-40c8-9234-11f8e97139a8.png">

Reply:
On the subject of the docs, it's probably also worth fixing the first order refinement criterion description.  From what I see, the code is not actually calculating

$$ \left \| \frac{d \ln q}{d \ln x} \right \|$$